code/processes/orchestrator.q

```diff
@@ -0,0 +1,56 @@
+/TorQ Orchestrator Process
+
+\d .orch
+
+/default parameters
+
+scalingdetails:([] time:`timestamp$(); procname:`$(); dir:`$(); instancecreated:`$(); instanceremoved:`$(); totalnumofinstances:`int$(); lowerlimit:`int$(); upperlimit:`int$());	/table for tracking scaling
+
+processlimitscsv:hsym first .proc.getconfigfile"processlimits.csv";	/location of csv file
+limits:1!("SII";enlist ",")0:processlimitscsv;	/table of scalable processes and the max number of instances allowed for each
+scaleprocslist:exec procname from limits;	/list of scalable processes
+
+/initialises connection to discovery process and creates keyed table containing the number of instances of each scalable process
+getscaleprocsinstances:{[] 
+	.servers.startup[];
+	`.orch.procs set string@/:exec procname from .servers.procstab;
+	`.orch.scaleprocsinstances set ([procname:scaleprocslist] instances:{sum procs like x,"*"}@/:string@/:scaleprocslist);
+	}
+
+getscaleprocsinstances[];
+
+/function to scale up a process
+scaleup:{[procname]
```

Comment:
Again scaleup and scaledown here there looks to be a lot of duplication in the code.  Potentially parameterize or project.